Core/src/org/sleuthkit/autopsy/apputils/ResetWindowsAction.java

```diff
@@ -51,46 +52,74 @@
+    private final static String CASE_TO_REOPEN_FILE = "caseToOpen.txt"; 
 
     @Override
     public boolean isEnabled() {
-        return !Case.isCaseOpen();
+        return true;
     }
 
-    @NbBundle.Messages({"ResetWindowAction.confirm.title=Reset Windows",
-        "ResetWindowAction.confirm.text=The program will close and restart to perform the resetting of window locations.\n\nAre you sure you want to reset all window locations?"})
+    @NbBundle.Messages({"ResetWindowAction.confirm.text=In order to perform the resetting of window locations the software will close and restart. "
+        + "If a case is currently open it will be closed. If ingest or a search is currently running it will be terminated. "
+        + "Are you sure you want to restart the software to reset all window locations?",
+        "ResetWindowAction.caseCloseFailure.text=Unable to close the current case, "
+        + "the software will restart and the windows locations will reset the next time the software is closed.",
+        "ResetWindowAction.caseSaveMetadata.text=Unable to save current case path, "
+        + "the software will restart and the windows locations will reset but the current case will not be opened upon restart."})
 
     @Override
     public void performAction() {
         SwingUtilities.invokeLater(() -> {
-            int response = JOptionPane.showConfirmDialog(
-                    WindowManager.getDefault().getMainWindow(),
-                    Bundle.ResetWindowAction_confirm_text(),
-                    Bundle.ResetWindowAction_confirm_title(),
-                    JOptionPane.YES_NO_OPTION,  JOptionPane.PLAIN_MESSAGE);
-            if (response == JOptionPane.YES_OPTION) {
+            boolean response = MessageNotifyUtil.Message.confirm(Bundle.ResetWindowAction_confirm_text());
+            if (response) {
+                //adding the shutdown hook, closing the current case, and marking for restart can be re-ordered if slightly different behavior is desired
                 Runtime.getRuntime().addShutdownHook(new Thread() {
                     @Override
                     public void run() {
                         try {
-                            FileUtils.deleteDirectory(new File(PlatformUtil.getUserConfigDirectory() + File.separator + "Windows2Local"));
+                            FileUtils.deleteDirectory(new File(PlatformUtil.getUserConfigDirectory() + File.separator + WINDOWS2LOCAL));
                         } catch (IOException ex) {
-                            logger.log(Level.WARNING, "Unable to delete config directory, window locations will not be reset.", ex);
+                            //While we would like the user to be aware of this in the unlikely event that the directory can not be deleted
+                            //Because our deletion is being attempted in a shutdown hook I don't know that we can pop up UI elements during the shutdown proces
```

Comment:
If I delete the file outside of the shut down hook it makes them again when it shut downs closing the top components. 

I have 0 clue how much of the UI is left at this point in the shut down process or even if the logging will work, I guess I can try to make it unable to be deleted by opening something in another folder, but I haven't confirmed how to create a problem only that it could potentially occur because of the exception the method throws.